src/run/ongoing.ts

```diff
@@ -13,19 +14,34 @@ export interface FinishedRun {
   assertions: Assertion[];
 }
 
+const MAX_CONSECUTIVE_ERRORS = 5;
+
 export const waitForRunEnd = async (client: ApiClient, startedRun: StartedRun): Promise<FinishedRun> => {
-  let runInfo: RunInformationResponse;
+  let runInfo: RunInformationResponse | undefined;
   let oldStatus: number = -1;
+  let errorCount = 0;
   do {
-    await setTimeout(5000); // Initial delay even on first iteration because run duration might not be populated yet
-    runInfo = await client.getRunInformation(startedRun.runId);
-    const statusMsg = `Run status is now ${statusName(runInfo.status)} [${runInfo.status}]`;
-    runInfo.status !== oldStatus ? core.info(statusMsg) : core.debug(statusMsg);
-    oldStatus = runInfo.status;
-    if (runInfo.injectStart > 0) {
-      await getAndLogMetricsSummary(client, runInfo);
+    try {
+      await setTimeout(5000); // Initial delay even on first iteration because run duration might not be populated yet
```

Comment:
Can't you poll until you get it? What if it's still not there after 5s?

Reply:
1. This is in  a loop, we already poll every 5 seconds, and we tolerate up to the 5th consecutive error, so we can wait 25 seconds really.
2. In the existing CI plugins, this is just a 1 second initial delay.

BTW, I assume that must only really be relevant for the self-hosted version, I don't see why we would have this issue with the cloud code base? 🤔 But I don't see much point in _not_ waiting 5 seconds for the first polling attempt anyway (we're really waiting for the run to start injecting at the beginning, which takes a bit of time).

Reply:
Oh, did not see the loop...